share/functions/funced.fish

```diff
@@ -122,8 +130,50 @@ function funced --description 'Edit function definition'
                     continue
                 end
                 echo (_ "Cancelled function editing")
+            else if test -n "$writepath"
+                if not set -q _flag_save
+                    echo (_ "Warning: the file containing this function has not been saved. Changes will be lost when fish is closed.")
+                    set -l prompt (printf (_ 'Save function to %s\? [Y/n]') "$writepath")
+                    read -p "echo $prompt\  " response
```

Comment:
This can just be `read -P "$prompt "` (`--prompt-str`)

Reply:
(or even `read -p "printf (_ 'Save function to %s\? [Y/n]') \"$writepath\""`, or `read -p "$(printf (_ 'Save function to %s\? [Y/n]') "$writepath")"`